api/providers/tasks.py

```diff
@@ -453,10 +453,16 @@ def handle_registration_row(row, initiator, provider, schema, auto_approval=Fals
         )
         # Remove all contributors except the initiator if created from an existing node
         if node:
+            # Temporarily make initiator contributor visible so that removal of the others can succeed.
+            initiator_contributor = draft.contributor_set.get(user=initiator)
+            if not initiator_contributor.visible:
```

Comment:
This has to happen before the `for` loop below since initiator contributor could after all visible ones in the iteration.